File: lec9.py
def string_compression(string):
    n = len(string)
    i = 1
    ans = string[0]

    while i < n:
        while i < n and ans[-1] == string[i]:
            i += 1

        if i < n:
            ans += string[i]
            i += 1

    print(ans)


def string_compression_counts(string):
    n = len(string)
    i = 1
    count = 0
    ans = string[0]

    while i < n:
        count = 1

        while i < n and ans[-1] == string[i]:
            i += 1
            count += 1

        if count >= 1:
            ans = ans + ":" + str(count) + "\n"

        if i < n:
            ans += string[i]

        i += 1

    print(ans)


def count_of_hi(s):
    length = len(s)
    i = 0
    count = 0

    s = s.lower()

    while i < length - 1:
        if s[i] == 'h' and s[i + 1] == 'i':
            count += 1
            i += 2
        else:
            i += 1

    print(count)
    
    
def count_of_hi_in_hit(s):
    length = len(s)
    i = 0
    count = 0

    while i < length - 1:
        if s[i] == 'h' and s[i + 1] == 'i':

            if i + 2 < length and s[i + 2] == 't':
                i += 3
            else:
                count += 1
                i += 2

        else:
            i += 1

    return count


def remove_hi(s):
    str1 = ""
    length = len(s)
    i = 0

    while i < length:

        if i + 1 < length and s[i] == 'h' and s[i + 1] == 'i':

            if i + 2 < length and s[i + 2] == 't':
                str1 += "hi"
                i += 2
            else:
                i += 2

        else:
            str1 += s[i]
            i += 1

    print(str1)
    
    
def remove_hit(s):
    str1 = ""
    length = len(s)
    i = 0

    while i < length:

        if (i + 2 < length and
            s[i] == 'h' and
            s[i + 1] == 'i' and
            s[i + 2] == 't'):

            i += 3

        else:
            str1 += s[i]
            i += 1

    print(str1)

File: lec9.java
import java.util.*;

public class lec9 {
    public static Scanner scn = new Scanner(System.in);

    public static void stringCompression(String str) {
        int n = str.length(), i = 1;
        String ans = str.charAt(0) + "";
        while (i < n) {
            while (i < n && ans.charAt(ans.length() - 1) == str.charAt(i)) {
                i++;
            }
            if (i < n) {
                ans += str.charAt(i);
                i++;
            }
        }
        System.out.print(ans);
    }

    public static void stringCompressionCounts(String str) {
        int n = str.length(), i = 1, count = 0;
        String ans = str.charAt(0) + "";
        while (i < n) {
            count = 1;
            while (i < n && ans.charAt(ans.length() - 1) == str.charAt(i)) {
                i++;
                count++;
            }

            if (count >= 1) {
                ans = ans + ":" + count + "\n";
            }
            if (i < n) {
                ans += str.charAt(i);
            }
            i++;
        }
        System.out.print(ans);
    }

    public static void countOfHi(String str) {
        int len = str.length(), i = 0, count = 0;
        str = str.toLowerCase();

        while (i < len - 1) {
            if (str.charAt(i) == 'h' && str.charAt(i + 1) == 'i') {
                count++;
                i += 2;
            } else {
                i++;
            }
        }
        System.out.println(count);
    }

    public static int countOfHiInHit(String str) {
        int len = str.length(), i = 0, count = 0;

        while (i < len) {
            if (str.charAt(i) == 'i' && str.charAt(i - 1) == 'h') {
                if (i + 1 < len && str.charAt(i + 1) == 't') {
                    i += 2;
                } else {
                    i++;
                    count++;
                }
            } else {
                i++;
            }
        }
        return count;
    }

    public static void removeHi(String str) {
        String str1 = "";
        int len = str.length(), i = 0;

        while (i < len) {
            if (str.charAt(i) == 'h' && str.charAt(i + 1) == 'i') {
                if (i + 1 < len && str.charAt(i + 2) != 't') {
                    i += 2;
                } else {
                    str1 += str.charAt(i);
                    i++;
                }
            } else {
                str1 += str.charAt(i);
                i++;
            }
        }
        System.out.println(str1);
    }

    public static void removeHit(String str) {
        String str1 = "";
        int len = str.length(), i = 0;

        while (i < len) {
            if (i + 2 < len && str.charAt(i) == 'h' && str.charAt(i + 1) == 'i' && str.charAt(i + 2) == 't') {
                i += 3;
            } else {
                str1 += str.charAt(i);
                i++;
            }
        }
        System.out.println(str1);
    }

    public static void main(String[] args) {
        stringCompressionCounts(scn.nextLine());
    }
}